client_monitor.go

```diff
@@ -57,6 +57,8 @@ func (m *clientMonitor) Start(ctx context.Context) error {
 // uninitialized.  Unlike SetProducerStatus, it does not broadcast the change
 // and is only meant to be used during initial client startup.
 func (m *clientMonitor) InitializeProducerStatus(queueName string) {
+	m.statusSnapshotMu.Lock()
+	defer m.statusSnapshotMu.Unlock()
```

Comment:
Found I needed to add this mutex when trying to run my new stress test. Most other client monitor functions already lock it.

Reply:
Makes sense. As this wasn't dynamic before it wasn't really needed.